notifique/src/main/java/com/nathanrassi/notifique/NotifiqueActivity.kt

```diff
@@ -29,33 +31,20 @@ class NotifiqueActivity : AppCompatActivity() {
     checkNotificationPermission()
   }
 
-  override fun onActivityResult(
-    requestCode: Int,
-    resultCode: Int,
-    data: Intent?
-  ) {
-    super.onActivityResult(requestCode, resultCode, data)
-    if (requestCode == notificationPermissionRequestCode) {
-      checkNotificationPermission()
-    }
-  }
-
   private fun checkNotificationPermission() {
     if (!NotificationManagerCompat.getEnabledListenerPackages(this).contains(packageName)) {
+
       val notificationSnackbar = Snackbar.make(
           findViewById(android.R.id.content),
           R.string.snackbar,
           Snackbar.LENGTH_INDEFINITE
       )
-      notificationSnackbar.setAction(R.string.snackbar_action) {
-        val intent = Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS)
-        if (packageManager.queryIntentActivities(intent, 0).isEmpty()) {
-          // TODO: rare problem.
-        } else {
-          startActivityForResult(intent, 0)
+      class SnackbarNotificationListener : View.OnClickListener {
```

Comment:
why the named class? i like the kotlin syntax for the anonymous inner class as the last parameter. see above existing code.

Reply:
b/c it was easier.  
ill change that